src/tests/CredentialsTest.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+use AWS\CRT\Auth\AwsCredentials as AwsCredentials;
+use AWS\CRT\Auth\StaticCredentialsProvider as StaticCredentialsProvider;
+
+require_once('common.inc');
+
+final class CredentialsTest extends CrtTestCase {
+
+    public function testEmptyCredentials() {
+        $this->expectException(InvalidArgumentException::class);
+        $creds = new AwsCredentials(AwsCredentials::defaults());
+        $this->assertNotNull($creds, "Failed to create default/empty credentials");
+        $creds = null;
+    }
+
+    private function credentialsOptions() {
+        $options = AwsCredentials::defaults();
+        $options['access_key_id'] = 'TESTAWSACCESSKEYID';
+        $options['secret_access_key'] = 'TESTSECRETaccesskeyThatDefinitelyDoesntWork';
+        $options['session_token'] = 'ThisIsMyTestSessionTokenIMadeItUpMyself';
+        $options['expiration_timepoint_seconds'] = 42;
```

Comment:
Is this expired for the purposes of the test or not expired?

Reply:
It doesn't actually matter for this test, though if I were going to test expiration, I'd set it to a time in the future, then wait past that time. Our auth tests handle that case.